content/articles/understanding-when-to-use -jsp-and-when-to-use-servlet/index.md

```diff
@@ -0,0 +1,184 @@
+
+```java
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class ClassName extends HttpServlet {
+	// Class variables are declared here
+	protected void Service(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		{
+			// Local variables are declared here
+			// This is where you will write your Java code.
+			PrintWriter out = response.getWriter();
+			out.print("Hello world.");
+		}
+
+	}
+}
+```
+JSP code, on the other hand, does not necessitate the creation of a class and Servlet objects such as `PrintWriter` because the web server, in our case *Tomcat*, does it as shown:
+
+```jsp
+ <body>
+  <%
+  out.print("Hello World");
+  %>
+</body>
+
+```
+#### Process of converting JSPs to Servlets
+
+1. Tomcat offers the `HttpServlet` parent class, which is extended by the newly formed class. The name of the Servlet class is determined by concatenating the JSP file name with the `.jsp` extension.
+2. Tomcat or any equivalent web server provides the `_jspService`, `doPost` or `doGet` method, as well as the `request` and `response` parameters as shown:
+   ```Java
+   public void _jspService(HttpServletRequest request, HttpServletResponse response)                        
+   {
+      PrintWriter out = response.getWriter();
+      response.setContentType("text/html");
+      out.write("<html><body>");
+      out.write("Hello world");
+      out.write("</body></html>");
+   }
+   ```
+3. The JSP code must be enclosed in one of the tags listed below
+- Scriptlet tag(<% code %>) 
+- declaration tag(<%! declaration %>)
+- Directive Tag(<%@ page ... %>)
+- Expression tag(<%= output%> )
+
+We will presume that you are familiar with JSP tags and will not describe them in this article. The JSP code contained within the following JSP tags will be copied into the Servlet page, as follows: 
+- The content of the scriptlet tag will be copied into the corresponding Servlet `_jspService()` method
+- The content of the declaration tag will be copied outside of the associated Servlet page's `_jspService()` method, and
+- All Java imports are contained within directive tags in JSP, and the contents of these tags are copied outside of the resulting `Servlet` class
+  
+### Example
+To explain how JSP gets converted to the matching `Servlet` during code execution, we'll use some simple JSP code.
+
+#### Steps to take
+
+#### Step 1: Create a new project
+Using Netbeans IDE click on File -> New Project -> Java Web -> Web Application and select next. Last but not least, give the project a name and click finish. In my case, I'll call it **JSPToServlet**. Finally, choose an appropriate server and click finish. In my case, I'll be using Tomcat version 9. 

```

Comment:
You said at the beginning of this article that you will be using Eclipse but you are now giving instructions for Netbeans. This is confusing.